x-pack/legacy/plugins/infra/public/components/waffle/node_context_menu.tsx

```diff
@@ -43,83 +48,136 @@ export const NodeContextMenu = ({
 }: Props) => {
   const uiCapabilities = useKibana().services.application?.capabilities;
   const inventoryModel = findInventoryModel(nodeType);
+  const nodeDetailFrom = currentTime - inventoryModel.metrics.defaultTimeRangeInSeconds * 1000;
   // Due to the changing nature of the fields between APM and this UI,
   // We need to have some exceptions until 7.0 & ECS is finalized. Reference
   // #26620 for the details for these fields.
   // TODO: This is tech debt, remove it after 7.0 & ECS migration.
   const apmField = nodeType === 'host' ? 'host.hostname' : inventoryModel.fields.id;
 
-  const nodeLogsMenuItem = {
-    name: i18n.translate('xpack.infra.nodeContextMenu.viewLogsName', {
-      defaultMessage: 'View logs',
+  const showDetail = inventoryModel.crosslinkSupport.details;
+  const showLogsLink =
+    inventoryModel.crosslinkSupport.logs && node.id && uiCapabilities?.logs?.show;
+  const showAPMTraceLink =
+    inventoryModel.crosslinkSupport.apm && uiCapabilities?.apm && uiCapabilities?.apm.show;
+  const showUptimeLink =
+    inventoryModel.crosslinkSupport.uptime && (['pod', 'container'].includes(nodeType) || node.ip);
+
+  const inventoryId = useMemo(() => {
+    switch (nodeType) {
+      case 'host':
+        if (node.ip) {
+          return { label: <EuiCode>host.ip</EuiCode>, value: node.ip };
+        } else {
+          return { label: '', value: '' };
+        }
+      case 'pod':
+        return { label: <EuiCode>pod.uid</EuiCode>, value: node.id };
+      case 'container':
+        return { label: <EuiCode>container.id</EuiCode>, value: node.id };
+      case 'awsEC2':
+      case 'awsRDS':
+      case 'awsSQS':
+      case 'awsS3':
+        return { label: <EuiCode>instance.id</EuiCode>, value: node.id };
```

Comment:
I wonder if we shouldn't be getting the fields from the inventory models instead of adding another place you have to edit for adding a new model. We could rewrite this to look something like:

```typescript
const inventoryId = useMemo(() => {
  const fields = findInventoryFields(nodeType, options.fields);
  if (nodeType === 'host') {
    if (node.ip) {
      return { label: <EuiCode>host.ip</EuiCode>, value: node.ip };
     } else {
      return { label: '', value: '' };
     }
  }
  return { label: <EuiCode>{field.id}</EuiCode>, value: node.id };
}, [nodeType, node, options.fields]);
```
This way we are only adding the edge cases when needed which for the time being will probably only be the `host` inventory type.

Reply:
I'd be down with that. `id` is a common concept across all inventory, so that makes sense.